### DISCUSSION TITLE
Hide Submit and Cancel Buttons in Resource wizard

### DISCUSSION THREAD

**Discussion by emmexx** [filamentphp/filament#8765] [Help] [answered]
I'd like to hide the submit and cancel buttons that are below the frame of the wizard for a resource.
The manual (v. 3.x) says:

> You may use the submitAction() method to render submit button HTML or a view at the end of the wizard, on the last step. This provides a clearer UX than displaying a submit button below the wizard at all times:

[Rendering a submit button on the last step](https://filamentphp.com/docs/3.x/forms/layout/wizard#rendering-a-submit-button-on-the-last-step)

But the manual gives no clue on how to hide the buttons below the wizard.

The only simple way I found is css, adding hidden to the `fi-form-actions` div but I don't like that.

Apart from customizing the view, is there a simple way to accomplish that?

**Comment by leandrocfe** [comment]
If you are using on resources, you should use [HasWizard](https://filamentphp.com/docs/3.x/panels/resources/creating-records#using-a-wizard) trait.

**Reply by emmexx** [reply]
The wizard is in a Relation Manager `public static function form(Form $form): Form {...`

**Reply by leandrocfe** [reply]
I think you can hide actions using `->createAnother(false)->modalSubmitAction(false)`

**Reply by emmexx** [reply]
I'm not using a modal, it is a wizard form for the main Model in a Relation Manager.

**Reply by leandrocfe** [reply]
Share your whole code please

**Reply by emmexx** [reply]
AccommodationResource file created with `php artisan make:filament-relation-manager AccommodationResource ...`

```
<?php

namespace App\Filament\myPanel\Resources;

use App\Filament\myPanel\Resources\AccommodationResource\Pages;
use App\Filament\myPanel\Resources\AccommodationResource\RelationManagers;
use App\Models\Accommodation;
use Filament\Forms;
use Filament\Forms\Components\Placeholder;
use Filament\Forms\Contracts\HasForms;
use Filament\Forms\Form;
use Filament\Forms\FormsComponent;
use Filament\Resources\RelationManagers\RelationManager;
use Filament\Resources\Resource;
use Filament\Tables;
use Filament\Tables\Columns\TextColumn;
use Filament\Tables\Table;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\SoftDeletingScope;
use Illuminate\Validation\Rule;
use Illuminate\Support\HtmlString;
use Illuminate\Support\Facades\Blade;

class AccommodationResource extends Resource
{
    protected static ?string $model = Accommodation::class;

    public static function form(Form $form): Form
    {
        return $form
            ->columns(1)
            ->schema([

                Forms\Components\Wizard::make([
                    Forms\Components\Wizard\Step::make('step 1')
                    ->schema([
                        Forms\Components\TextInput::make('name'),
                    ]),
                    Forms\Components\Wizard\Step::make('step 2')
                    ->schema([
                        Forms\Components\TextInput::make('address'),
                    ]),
                ])
                ->submitAction(new HtmlString('<button type="submit">Submit</button>'))
         ]);
   }

 ...
}
```

**Reply by leandrocfe** [reply, marked as answer]
Ok, this will create a `xxRelationManger.php` file that you have a `createAction` `editAction`..
So you can use steps

```php
->headerActions([
    Tables\Actions\CreateAction::make()
        ->steps([
            Step::make('step1')
                ->schema([
                    Forms\Components\TextInput::make('field_1')
                ]),
            Step::make('step2')
                ->schema([
                    Forms\Components\TextInput::make('field_2')
                ])
               ...
        ]),
])
```

**Reply by emmexx** [reply]
I don't understand why the difference between the resource way and the pages/create way but it works.

Thank you

**Comment by mauricoder** [comment]
Just adding a note here for future people looking for this. If you have a Wizard on a Resource form, you can disable Cancel and Save buttons that appear bellow the wizard by commenting out the Actions in the getFormActions method for the Edit and Create pages. For example, let's say your resource is UserResource, then you'll have a UserResource/Pages/CreateUser.php and UserResource/Pages/EditUser.php files.

On each one of these you can do this edit to disable buttons bellow the Wizard:

```
    /**
     * @return array<Action | ActionGroup>
     */
    protected function getFormActions(): array
    {
        return [
            //$this->getSaveFormAction(),
            // $this->getCancelFormAction(),
        ];
    }
```

**Reply by reasecret** [reply]
I'm looking for this solution. Thanks...
